Firestore/Source/API/FIRQuery.m

```diff
@@ -256,6 +256,82 @@ - (FIRQuery *)queryWhereFieldPath:(FIRFieldPath *)path isGreaterThanOrEqualTo:(i
                                  value:value];
 }
 
+- (FIRQuery *)queryWherePredicate:(NSPredicate *)predicate {
```

Comment:
This method is very large. Consider breaking it down into a few smaller pieces.

For example, you could make `queryWherePredicate:` do the instanceof checks and then delegate to strongly-typed single-purpose methods like `queryWhereComparisonPredicate:` and `queryWhereCompoundPredicate`.

Reply:
done